internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,257 @@
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	//TODO(varun359): Get build information from the manifest and task definition for workloads
+
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		return o.validateEnvName()
+	}
+	envs, err := o.getDeployedEnvironments()
+	switch {
+	case err != nil:
+	case len(envs) == 0:
+	case len(envs) == 1:
+	}
+	switch {
+	case err != nil:
+		return err
+	case len(envs) == 0:
+		return fmt.Errorf("no deployed environments found in the app %s", o.appName)
+	case len(envs) == 1:
+		log.Infof("Only one environment found, defaulting to: %s\n", color.HighlightUserInput(envs[0]))
+		o.envName = envs[0]
+		return nil
+
+	}
+	selectedEnvName, err := o.prompt.SelectOne("Select an environment in which you want to test", "", envs, prompt.WithFinalMessage("Environment:"))
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = selectedEnvName
+	return nil
+}
+
+func (o *localRunOpts) getDeployedEnvironments() ([]string, error) {
+	envConfig, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("get environments for app %s: %w", o.appName, err)
+	}
+
+	var deployedEnvs []string
+	for _, env := range envConfig {
+		isDeployed, err := o.isEnvironmentDeployed(env.Name)
+		if err != nil {
+			return nil, err
+		}
+		if isDeployed {
+			deployedEnvs = append(deployedEnvs, env.Name)
+		}
+	}
+
+	return deployedEnvs, nil
+}
+
+func (o *localRunOpts) isEnvironmentDeployed(envName string) (bool, error) {
+	var checker versionCompatibilityChecker
+
+	checker, err := o.envVersionChecker(envName)
+	if err != nil {
+		return false, err
+	}
+
+	currVersion, err := checker.Version()
+	if err != nil {
+		return false, fmt.Errorf("get environment %q version: %w", envName, err)
+	}
+	if currVersion == version.EnvTemplateBootstrap {
+		return false, nil
+	}
+	return true, nil
+}
+func (o *localRunOpts) validateEnvName() error {
+	isDeployed, err := o.isEnvironmentDeployed(o.envName)
+	if err != nil {
+		return err
+	}
+	if !isDeployed {
+		return fmt.Errorf(`cannot use an environment which is not deployed Please run "copilot env deploy, --name %s" to deploy the environment first`, o.envName)
+	}
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskWorkloadName() error {
+	if o.wkldName != "" {
+		return o.validateWkldName()
+	}
+	services, err := o.deployStore.ListDeployedServices(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("Get services: %w", err)
+	}
+
+	jobs, err := o.deployStore.ListDeployedJobs(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("Get jobs: %w", err)
+	}
+
+	workloads := append(services, jobs...)
+	if len(workloads) == 0 {
+		return fmt.Errorf("no workloads found in this environment %s", o.envName)
+	}
+	if len(workloads) == 1 {
+		log.Infof("Only one workload found in this environment, defaulting to: %s\n", color.HighlightUserInput(workloads[0]))
```

Comment:
I'm starting to wonder if the prompt flow makes sense 🤔 . An alternative that I can think of is to reverse the env-wkld order. 
```
$ copilot local run
What workload do you want to test? web
Select an environment in which you want to test "web":
   > test
   > prod
```
What are our thoughts on this?

Reply:
Yes, we had a discussion on this earlier. To get the deployed workloads. We should iterate through all the envs to get them and then when the user selects the workload, we need to get all the environments in which the workload is deployed in and prompt to the user.

````
envs := getAllDeployedEnvs() #-step1
for env in envs:
  wklds.add(env.listServices())
  wklds.add(env.listJobs())

#prompt to select a deployed service
if service in multiple envs:
  #prompt to select env for service
````

Even this approach works but In this case doesn't it increase the latency(delay) even more after doing this? Since we are doing the step 1 in the above code already. Isn't it better to prompt the user the deployedEnvs first (instead of iterating through it and get the deployed workloads) and then get deployedWorkloads(there is no difference between both I guess)?